### PR TITLE
fix cart + checkout

### PR DESCRIPTION
**Pull request by lyninx**
fixing how products are added, updated, and removed form the cart, as well as how a checkout is generated.

since there is a perceptible delay when updating items in the cart, my next PR will contain performance improvements (caching + client side calculations), for modifying the cart.

### CODE REVIEW

File: src/components/cart.js [hunk 1]
```diff
@@ -77,8 +78,8 @@ export default class Cart extends Component {
       const data = merge(lineItem, this.options.viewData);
       data.classes = this.classes;
       data.lineItemImage = this.imageForLineItem(data);
-      data.variantTitle = data.variant_title === 'Default Title' ? '' : data.variant_title;
-      data.formattedPrice = formatMoney(data.line_price, this.globalConfig.moneyFormat);
+      data.variantTitle = data.variant.title === 'Default Title' ? '' : data.variant.title;
+      data.formattedPrice = formatMoney(data.variant.price * data.quantity, this.globalConfig.moneyFormat);
```
Comment (minasmart) [inline src/components/cart.js]: this should rely on the shop resource. Is that's what's driving `globalConfig` now?

File: test/unit/cart.js [hunk 2]
```diff
@@ -67,7 +70,8 @@ describe('Cart class', () => {
             title: 'test',
             variant_title: 'test2',
             line_price: 20,
-            quantity: 1
+            quantity: 1,
+            variant: { image: { src: 'cdn.shopify.com/image.jpg' } }
```
Comment (minasmart) [inline test/unit/cart.js]: use braces consistently. Nowhere else is are you doing `{ stuff` it's always `{stuff`. This should probably be in the linter.

Comment (lyninx) [inline test/unit/cart.js]: ah. i always default to adding spaces, and then fix it when the linter yells at me. i'm surprised it didn't catch this.

File: src/components/cart.js (hunk 1 above)
Comment (lyninx) [inline src/components/cart.js]: gotchya. i'll look into updating this.

**Comment by lyninx** [comment]
i'll be sure to take care of fixing linting of tests in a future PR

**Pull request merged**